pkg/operator/profilecalculator.go

```diff
@@ -626,6 +630,46 @@ func (pc *ProfileCalculator) getNodePoolNameForNode(node *corev1.Node) (string,
 	return nodePoolName, nil
 }
 
+// TunedRecommend returns a name-sorted TunedProfile slice out of
+// a slice of Tuned objects.
+func TunedProfiles(tunedSlice []*tunedv1.Tuned) []tunedv1.TunedProfile {
+	tunedProfiles := []tunedv1.TunedProfile{}
+	m := map[string]tunedv1.TunedProfile{}
+
+	for _, tuned := range tunedSlice {
+		if tuned.Spec.Profile != nil {
+			for _, v := range tuned.Spec.Profile {
+				if v.Name != nil && v.Data != nil {
+					if existingProfile, found := m[*v.Name]; found {
+						if *v.Data == *existingProfile.Data {
+							klog.Infof("duplicate profiles names %s but they have the same contents", *v.Name)
+						} else {
+							klog.Errorf("ERROR: duplicate profiles named %s with different contents found in Tuned CR %q", *v.Name, tuned.Name)
+						}
+					}
+					m[*v.Name] = v
+				}
+			}
+		}
+	}
+	for _, tunedProfile := range m {
+		if tunedProfile.Name == nil {
+			// This should never happen (openAPIV3Schema validation); ignore invalid profiles
+			continue
```

Comment:
we guard against this few lines above in the same function; furthermore, I'm struggling to find a meaningful log entry here (ordering in the slice is no longer relevant, and the profile is anonymous in the first place). So I'd drop this check from the flattening loop